### ISSUE TITLE
Pacific Northwest National Lab - Issues Mega Thread

### ISSUE THREAD

**Issue by LouisJenkinsCS** [chapel-lang/chapel#10176]
As per @bradcray's suggestion, I will implement a list of issues that are impeding progress of the project of creating HyperGraph data structures and abstractions

# Urgent

- [x] Compiler shows type mismatch in assignment to same type: #10358
- [x] Compiler error that occurs from casting `c_void_ptr` to `c_ptr(?t)`: #10310 

# Important

- [x] Querying locale information causes large amounts of communication: #10747
- [ ] Array Aliasing and reducing communication: #10088, #10160 
- [x] Allowing user-types to make use of remote-value forwarding: #9717 

# Low Priority

- [ ] Tools to eliminate false sharing: #10771 
- [x] Querying locale information from distributed arrays causes communication: #10528 
- [x] Compiler throws 'ambiguous call' when accessing type of forwarded classes: #10355
- [x] Creating arrays larger than size 2MB inside local block causes errors: #10536
- [ ] Atomics are suffering from severe performance issues under intrinsics: #10996, #10998

# Nice to have

- [ ] Docker image for master:  #10510
- [ ] `start_test --performance` with multiple locale configurations: #9596 
- [ ] Weighted Random Sampling (@hmedal): #9805 
- [ ] Reducing Stack Size of Tasks: #9984 

# Discussions

- [ ] Design and Proposal for Chapel Aggregation Library: #10386 
- [x] Data Aggregation and Coalescing: #9727

**Comment by hmedal** [comment]
*Nice to Have:* Weighted random sampling - Issue #[XXX](https://github.com/chapel-lang/chapel/issues/9805)

**Comment by LouisJenkinsCS** [comment]
Shifted issues on performance down outside of 'Urgent' to 'Important'; reserving 'Urgent' for actual issues that prevent forward progress.

**Comment by LouisJenkinsCS** [comment]
Closed issue on aggregation and coalescing because I created my own library for it that works wonderfully so far.

**Comment by tjstavenger-pnnl** [comment]
Also #10510 for a Chapel master Docker image and #9730 for a test framework.

**Comment by LouisJenkinsCS** [comment]
Added new issue: #10528, which is causing a significant degradable in performance, by at least an order of magnitude.

**Comment by LouisJenkinsCS** [comment]
Moved #10528 down from Urgent to Low Priority as I'll just have to make due with a workaround.

**Comment by LouisJenkinsCS** [comment]
Closed #10528 as the issue strangely no longer occurs... might be due to pulling a more recent version of master and clobbering.

**Comment by LouisJenkinsCS** [comment]
Closed issue #9717 as @bradcray added remote-value forwarding for user-defined types.

**Comment by LouisJenkinsCS** [comment]
Opened issue #10747

**Comment by LouisJenkinsCS** [comment]
Opened issue #10771 for the elimination of false sharing and preventing the compiler from eliminating class fields?

**Comment by LouisJenkinsCS** [comment]
Added issues #10996 and #10998 as they both effect performance current and future algorithms.

**Note: I am still an employee of PNNL and am currently a part-time external collaborator**

**Comment by bradcray** [comment]
w.r.t. the atomics issues and my trying to understand their nature:  If you're using cstdlib atomics and gnu (say), do you have problems?  (i.e., is this a workable solution?).  I saw your comment on making your code dummy-proof and proposed using CHPL_* checks within the code to warn such users about not being in the preferred configuration.  Thanks.

**Comment by bradcray** [comment]
@LouisJenkinsCS : I was reviewing where we were with these items this morning and wondering whether you'd consider [the progress Elliot reported](https://github.com/chapel-lang/chapel/issues/11029#issuecomment-454168012) on issue #11029 sufficient to check or downgrade in priority the "Atomics are suffering from severe performance issues" checkbox.  As I mentioned earlier today on #10996, we're keeping that issue open as a placeholder for the remaining optimization opportunity for intrinsics, but would prefer to put it off it that specific issue wasn't of the highest priority for you (which was my takeaway from the discussion on it).  Thanks.

**Comment by LouisJenkinsCS** [comment]
@bradcray Done. Moved to low priority and revised heading to mention that the issue is explicitly under intrinsics.

**Comment by bradcray** [comment]
I'm going to close this issue since it was designed primarily to prioritize tasks that PNNL was blocked on while actively working on CHGL and has not been maintained in recent years.  The underlying issues will remain open, I'm just closing this prioritization meta-issue.
